src/include/OpenImageIO/paramlist.h

```diff
@@ -163,52 +157,19 @@ class OIIO_API ParamValue {
 
 
 /// A list of ParamValue entries, that can be iterated over or searched.
-///
-class OIIO_API ParamValueList {
-    typedef std::vector<ParamValue> Rep;
+/// It's really just a std::vector<ParamValue>, but with a few more handy
+/// methods.
+class OIIO_API ParamValueList : public std::vector<ParamValue> {
 public:
     ParamValueList () { }
 
-    typedef Rep::iterator        iterator;
-    typedef Rep::const_iterator  const_iterator;
-    typedef ParamValue           value_type;
-    typedef value_type &         reference;
-    typedef const value_type &   const_reference;
-    typedef value_type *         pointer;
-    typedef const value_type *   const_pointer;
-
-    iterator begin () { return m_vals.begin(); }
-    iterator end () { return m_vals.end(); }
-    const_iterator begin () const { return m_vals.begin(); }
-    const_iterator end () const { return m_vals.end(); }
-    const_iterator cbegin () const { return m_vals.begin(); }
-    const_iterator cend () const { return m_vals.end(); }
-
-    reference front () { return m_vals.front(); }
-    reference back () { return m_vals.back(); }
-    const_reference front () const { return m_vals.front(); }
-    const_reference back () const { return m_vals.back(); }
-
-    reference operator[] (int i) { return m_vals[i]; }
-    const_reference operator[] (int i) const { return m_vals[i]; }
-    reference operator[] (size_t i) { return m_vals[i]; }
-    const_reference operator[] (size_t i) const { return m_vals[i]; }
-
-    void resize (size_t newsize) { m_vals.resize (newsize); }
-    size_t size () const { return m_vals.size(); }
-    bool empty () const { return size() == 0; }
-
     /// Add space for one more ParamValue to the list, and return a
     /// reference to its slot.
     reference grow () {
         resize (size()+1);
```

Comment:
I believe that was the goal of grow -- when you know you want one more, and are happy to do a few reallocations in order to guarantee you aren't wasting any slots.

But when you do want to amortize the cost are are ok with over-allocating, you can use emplace_back, which is now included since this class formally inherits from (rather than contains) std::vector.